carbon-design-system/carbon-addons-iot-react
# docs(table): add column customization section

Closes #3195

**Summary**
Created a new table story section for column customization. Refactored the mdx structure and removed
redundant stories. There are still old stories that use column customization knobs but once all feature areas have been covered we can remove the old stories.

Also used the following ideas that I would like to pitch for our stories in general:
- The stories follows the same capitalization pattern as carbon, i.e. all stories and all knobs start with a capital letter.
- The attribute `storyName` is the same as the name of the exported story const (in the .story.jsx file)` in CamelCase. This is in order for the URL to match the titles of the stories.
- A group of related stories links directly to the relevant documentation section. If possible and relevant, the stories in the story group match the different sub sections in the relevant documentation.
- A group of related stories should have a playground story

**Change List (commits, features, bugs, etc)**
- Created new section with a combination of new and existing stories 
- Updated the docs to match the new section and stories
- Removed WithOptionsToExploreColumnSettings from being exported, kept it for future debugging id needed.
- Created a new folder for the mdx files since they will grow in number for the table 


**Acceptance Test (how to verify the PR)**
1. Proof read the new and updated documentation
2. Verify that the [stories](https://deploy-preview-3205--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-column-customization--playground) make sense and that the knobs are working
3. Verify that the links to and within the mdx files are correct

**Regression Test (how to make sure this PR doesn't break old functionality)**
Existing unrelated documentation and stories are unaffected



**Things to look for during review**
